### PR TITLE
feat(core): set user-agent system properties

### PR DESCRIPTION
**Pull request by sktimalsina**
- [x] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*
- Update Kotlin SDK version to `0.17.6-beta`
- Change user agent singleton to set system properties for user agent string
- Changes to pinpoint as per new SDK

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [x] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [x] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by tjleing — APPROVED** [review]

File: core/src/main/java/com/amplifyframework/util/UserAgent.java [hunk 1]
```diff
@@ -56,29 +76,35 @@ public static synchronized void configure(@NonNull Map<Platform, String> platfor
...
-        // Pre-pend the additional platforms before Android user-agent
-        final StringBuilder userAgent = new StringBuilder();
-        for (Map.Entry<Platform, String> platform : platformVersions.entrySet()) {
-            userAgent.append(String.format("%s:%s ",
-                    platform.getKey().getLibraryName(),
-                    platform.getValue()));
+        HashMap<String, String> extras = new HashMap<>();
+        String libraryName;
+        String libraryVersion;
+        if (platformVersions.containsKey(Platform.FLUTTER)) {
```
Comment (div5yesh) [inline core/src/main/java/com/amplifyframework/util/UserAgent.java:82]: 🚀

File: core/src/main/java/com/amplifyframework/util/UserAgent.java [hunk 2]
```diff
@@ -56,29 +76,35 @@ public static synchronized void configure(@NonNull Map<Platform, String> platfor
...
+        instance = new UserAgent(libraryName, libraryVersion, extras);
+        instance.setMetaData();
+    }
 
-        instance = userAgent.toString();
+    /**
+     * AWS Kotlin SDK sets the user agent header by reading System properties.
+     * This method sets respective system properties for user agent metadata.
+     */
+    private void setMetaData() {
```
Comment (div5yesh) [inline core/src/main/java/com/amplifyframework/util/UserAgent.java:99]: 🙌

File: core/src/main/java/com/amplifyframework/util/UserAgent.java [hunk 3]
```diff
@@ -170,83 +218,11 @@ public enum Platform {
...
-            this.libraryName = sanitize(libraryName);
-            return this;
-        }
-
-        Builder libraryVersion(String libraryVersion) {
-            this.libraryVersion = sanitize(libraryVersion);
-            return this;
-        }
-
-        Builder systemName(String systemName) {
```
Comment (div5yesh) [inline core/src/main/java/com/amplifyframework/util/UserAgent.java:201]: does system name and version not add any value to us?

File: core/src/main/java/com/amplifyframework/util/UserAgent.java [hunk 4]
```diff
@@ -92,39 +118,61 @@ public static synchronized void reset() {
...
-                .libraryName(Platform.ANDROID.getLibraryName())
-                .libraryVersion(BuildConfig.VERSION_NAME)
-                .deviceManufacturer(Build.MANUFACTURER)
-                .deviceName(Build.MODEL)
-                .userLanguage(System.getProperty("user.language"))
-                .userRegion(System.getProperty("user.region"))
-                .toString();
+    @NonNull
+    @Override
+    public String toString() {
```
Comment (div5yesh) [inline core/src/main/java/com/amplifyframework/util/UserAgent.java:153]: Im guessing we are planning to use this string for API category user agent which uses ktor and not Kotlin SDK, right?

**Review by div5yesh — APPROVED** [review]

File: core/src/main/java/com/amplifyframework/util/UserAgent.java (hunk 4 above)
Comment (sktimalsina) [reply core/src/main/java/com/amplifyframework/util/UserAgent.java:153]: Yes, that and for backward compatibility for APIs which use `UserAgent::string`

**Pull request merged**